### PR TITLE
Clean up CLI operation categories and names, and add documentation for CLI

### PR DESCRIPTION
**Pull request by fnothaft**
Builds on #1262.

* Removed CalculateDepth command.
* Clean up argument doc in TransformFeatures.
* Move OUTPUT to an option for Flagstat.
* Remove some output saving arguments in the View command that don't exactly make sense.
* Remove unused Vcf2ADAM/ADAM2Vcf -dict option and `org.bdgenomics.adam.cli.DictionaryCommand` trait.
* Remove WigFix2ADAM command.
* Move away from ParquetLoadSaveArgs.
* Move reads2coverage to actions.
* Reads2Coverage does not only write to Parquet, so clean up docs.
* Clean up fragments<->reads argument names.

### CODE REVIEW

File: docs/source/50_cli.md [hunk 1]
```diff
@@ -0,0 +1,491 @@
...
+  logging level to set for Parquet's loggers. Defaults to `severe`.
+* `-parquet_page_size N`: The page size in bytes to use when writing Parquet
+  files. Defaults to 1MB (1024 * 1024).
+* `-print_metrics`: If provided, prints the
+  [instrumentation](https://github.com/bigdatagenomics/utils#instrumentation)
+  metrics to the log when the CLI operation terminates.
+
+#### Legacy output options {#legacy-output}
+
+Several tools in ADAM support saving back to legacy genomics output formats. Any
```
Comment (heuermh) [inline docs/source/50_cli.md]: legacy genomics output formats, scandalous!  :)

File: docs/source/50_cli.md [hunk 2]
```diff
@@ -0,0 +1,491 @@
...
+  if it would reduce the number of partitions to fewer than the number of
+  Spark executors. This may have a substantial performance cost, and will
+  invalidate any sort order.
+
+`anno2adam` converts VCFs with annotated variants (i.e., the VCF INFO fields)
+into Parquet using the VariantAnnotation schema. `anno2adam` takes the same two
+required arguments as `vcf2adam`. `anno2adam` takes the [default
+options](#default-args), and one additional option:
+
+* `-current_db`: A path to an existing Parquet file of VariantAnnotations. These
```
Comment (heuermh) [inline docs/source/50_cli.md]: `-current_db` is a poorly named argument.

File: docs/source/50_cli.md [hunk 3]
```diff
@@ -0,0 +1,491 @@
...
+
+The printing tools provide some form of user readable view of an ADAM file.
+These commands are useful for both quality control and debugging.
+
+### print
+
+Dumps a Parquet file to either the console or a text file as
+[JSON](http://www.json.org). Takes one required argument:
+
+1. `FILE(S)`: The file paths to load. This must be a Parquet formatted file.
```
Comment (heuermh) [inline docs/source/50_cli.md]: &rarr; These must be a Parquet formatted files.

**Comment by fnothaft** [comment]
Addressed review comments.

**Comment by heuermh** [comment]
Thank you, @fnothaft!

**Comment by laserson** [comment]
Shucks, missed this PR.  I use the `wigfix2bed` CLI command in the Spark book.  Out of scope for ADAM?

**Comment by laserson** [comment]
I'll use BEDOPS, which seems to have a pre-written tool for this

**Comment by heuermh** [comment]
@laserson We removed the `wigfix2bed` because it felt too small for a top-level command.  We've also revamped feature support since, so there is no reason to limit export to BED format.

If wigfix is something useful to have, perhaps we should support it as a feature format, and then conversions to/from will be supported by the `transformFeatures` command.  Create an issue if you feel so.

**Comment by fnothaft** [comment]
+1 @heuermh

**Comment by laserson** [comment]
Nah, I've determined the path of maximum laziness is to just use another tool rather than revive the tool in ADAM 😛

**Comment by fnothaft** [comment]
SGTM!

**Pull request merged**